### PR TITLE
feat: LT-2146 - OIEL Country Contract Types

### PR DESCRIPTION
**Pull request by iichr**
- allows exporter to provide the type of contract for a destination on their OIEL application
  - applicable to:
     - OIELs which can have destinations specified - `military/dual use` only
     - Open Individual Trade Control applications
- new contract type form
- new destinations summary list page; allows for:
  - editing of the contract types specific to a given country
  - adding/removing countries as destinations

contract type selection

destination summary list page

### CODE REVIEW

File: applications/helpers/countries.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def get_countries_missing_contract_types(request, object_pk):
+    return [
+        entry["country"]
+        for entry in get_application_countries_and_contract_types(request, object_pk)
+        if not entry["contract_types"]
+    ]
+
+
+class ContractTypes:
```
Comment (a-bek) [inline applications/helpers/countries.py:13]: Maybe pop this in Enum and below: `ContractTypes(contract_type).value`

File: applications/views/locations.py [hunk 2]
```diff
@@ -122,7 +137,104 @@ def init(self, request, **kwargs):
...
+        self.data = (
+            {
+                "contract_types": data_for_current_country[0]["contract_types"],
+                "other_contract_type_text": data_for_current_country[0]["other_contract_type_text"],
+            }
+            if data_for_current_country
+            else {}
+        )
+
+        if current_country != "all":
```
Comment (a-bek) [inline applications/views/locations.py:196]: Can we store "all" somewhere?

File: applications/forms/countries.py [hunk 3]
```diff
@@ -18,6 +19,61 @@ def countries_form(application_id):
...
+def contract_type_per_country_form(current_country, country_name):
+    return Form(
+        title=contractTypeStrings.AddContractTypesForm.TITLE + country_name,
+        description=contractTypeStrings.AddContractTypesForm.DESCRIPTION,
+        questions=[
+            HiddenField("countries", current_country),
+            Checkboxes(
+                name="contract_types[]",
+                options=[
+                    Option(key="nuclear_related", value=contractTypeStrings.NUCLEAR_RELATED,),
```
Comment (a-bek) [inline applications/forms/countries.py:54]: With the enum, we can just generate all these Options

Comment (iichr) [reply applications/forms/countries.py]: Much nicer now agreed 🚀

File: applications/helpers/countries.py (hunk 1 above)
Comment (iichr) [reply applications/helpers/countries.py]: Done!

File: applications/views/locations.py (hunk 2 above)
Comment (iichr) [reply applications/views/locations.py]: moved to variables

**Pull request merged**